dsumaroka/jv-work-with-file

dsumaroka opened the pull request.

okuzan commented on src/main/java/core/basesyntax/WorkWithFile.java:
```diff
@@ -1,7 +1,48 @@
+        return builder.toString();
+    }
+
+    public void getStatistic(String fromFileName, String toFileName) {
+        String[] fromFile = readFromFile(fromFileName);
+        String toFile = createReport(fromFile);
```
this method should contain three calls of other methods: readFile() - returns content, generateReport() -(you pass contents here), writeToFile()-(you pass report and filename here ), this will be a good code decomposition.example:

        String data = readFile(fromFileName);
        String report = generateReport(data);
        writeToFile(toFileName, report);
        
Also, public methods should go before private ones

okuzan reviewed (changes requested):
Please resturcture your code so we can proceed with the review. All literals should be constants

okuzan commented on src/main/java/core/basesyntax/WorkWithFile.java:
```diff
@@ -1,7 +1,55 @@
+import java.nio.file.Files;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
```
```suggestion
    private static final String SUPPLY_KEYWORD = "supply";
    private static final String BUY_KEYWORD = "buy";
    private static final String RESULT_KEYWORD = "result";
    private static final String DELIMITER_REGEX = "\\W+";
```

okuzan commented on src/main/java/core/basesyntax/WorkWithFile.java:
```diff
@@ -1,7 +1,55 @@
+    }
+
+    private String createReport(String strings) {
+        String[] strings1 = strings.split("\\W+");
```
```suggestion
        String[] tokens = strings.split(DELIMITER_REGEX );
```
Please pay attention to the naming, it's important. Remember, that developers spend way more time reading code than writing it

okuzan commented on src/main/java/core/basesyntax/WorkWithFile.java:
```diff
@@ -1,7 +1,55 @@
+    private static final String RESULT = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String toFile = readFromFile(fromFileName);
```
```suggestion
        String fileContents = readFromFile(fromFileName);
```

okuzan reviewed (changes requested):
Good job overall!

dsumaroka commented:
Ok,  thanks for the advice) I'll do


Пт, 27 окт. 2023 г. в 20:38, Oleh Kuzan ***@***.***>:

> ***@***.**** requested changes on this pull request.
>
> Good job overall!
> ------------------------------
>
> In src/main/java/core/basesyntax/WorkWithFile.java
> <https://github.com/mate-academy/jv-work-with-file/pull/1217#discussion_r1374872764>
> :
>
> > +    private static final String SUPPLY = "supply";
> +    private static final String BUY = "buy";
> +    private static final String RESULT = "result";
>
> ⬇️ Suggested change
>
> -    private static final String SUPPLY = "supply";
> -    private static final String BUY = "buy";
> -    private static final String RESULT = "result";
> +    private static final String SUPPLY_KEYWORD = "supply";
> +    private static final String BUY_KEYWORD = "buy";
> +    private static final String RESULT_KEYWORD = "result";
> +    private static final String DELIMITER_REGEX = "\\W+";
>
> ------------------------------
>
> In src/main/java/core/basesyntax/WorkWithFile.java
> <https://github.com/mate-academy/jv-work-with-file/pull/1217#discussion_r1374873556>
> :
>
> > +        String toFile = readFromFile(fromFileName);
> +        String report = createReport(toFile);
> +        writeToFile(toFileName, report);
> +    }
> +
> +    private String readFromFile(String fileName) {
> +        File file = new File(fileName);
> +        try {
> +            return Files.readAllLines(file.toPath()).toString();
> +        } catch (IOException e) {
> +            throw new RuntimeException("Can`t read file " + fileName, e);
> +        }
> +    }
> +
> +    private String createReport(String strings) {
> +        String[] strings1 = strings.split("\\W+");
>
> ⬇️ Suggested change
>
> -        String[] strings1 = strings.split("\\W+");
> +        String[] tokens = strings.split(DELIMITER_REGEX );
>
> Please pay attention to the naming, it's important. Remember, that
> developers spend way more time reading code than writing it
> ------------------------------
>
> In src/main/java/core/basesyntax/WorkWithFile.java
> <https://github.com/mate-academy/jv-work-with-file/pull/1217#discussion_r1374873774>
> :
>
> >      public void getStatistic(String fromFileName, String toFileName) {
> +        String toFile = readFromFile(fromFileName);
>
> ⬇️ Suggested change
>
> -        String toFile = readFromFile(fromFileName);
> +        String fileContents = readFromFile(fromFileName);
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mate-academy/jv-work-with-file/pull/1217#pullrequestreview-1702263198>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A7UEH4CYEQ4E77IVBAYKAB3YBPWSDAVCNFSM6AAAAAA6S2THZSVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTOMBSGI3DGMJZHA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

okuzan commented on src/main/java/core/basesyntax/WorkWithFile.java:
```diff
@@ -1,7 +1,56 @@
+            }
+        }
+        result = supply - buy;
+        StringBuilder builder = new StringBuilder();
```
```suggestion
        StringBuilder report = new StringBuilder();
```

okuzan reviewed (approved):
Well done 👏

dsumaroka commented:
Thanks)

Пт, 27 окт. 2023 г. в 21:21, Oleh Kuzan ***@***.***>:

> ***@***.**** approved this pull request.
>
> Well done 👏
> ------------------------------
>
> In src/main/java/core/basesyntax/WorkWithFile.java
> <https://github.com/mate-academy/jv-work-with-file/pull/1217#discussion_r1374907898>
> :
>
> > +    }
> +
> +    private String createReport(String strings) {
> +        String[] tokens = strings.split(DELIMITER_REGEX);
> +        int supply = 0;
> +        int buy = 0;
> +        int result;
> +        for (int i = 1; i < tokens.length; i++) {
> +            if (tokens[i].equals(SUPPLY_KEYWORD)) {
> +                supply += Integer.parseInt(tokens[i + 1]);
> +            } else if (tokens[i].equals(BUY_KEYWORD)) {
> +                buy += Integer.parseInt(tokens[i + 1]);
> +            }
> +        }
> +        result = supply - buy;
> +        StringBuilder builder = new StringBuilder();
>
> ⬇️ Suggested change
>
> -        StringBuilder builder = new StringBuilder();
> +        StringBuilder report = new StringBuilder();
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mate-academy/jv-work-with-file/pull/1217#pullrequestreview-1702318549>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A7UEH4GNE5ISVYUDXOHSP3DYBP3RFAVCNFSM6AAAAAA6S2THZSVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTOMBSGMYTQNJUHE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
